internal/action/main_test.go

```diff
@@ -0,0 +1,291 @@
+		t.Fatal(err)
+	}
+
+	_, err = client.Workspaces.Read(ctx, config.Organization, config.Name)
+	if err == nil {
+		t.Fatal("workspace should not exist, and an error should be returned")
+	}
+
+	if err.Error() != "resource not found" {
+		t.Fatalf("Error is not workspace not found: %s", err)
+	}
+
+	Run(config)
+
+	ws, err := client.Workspaces.Read(ctx, config.Organization, config.Name)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, ws.Name, config.Name)
+
+	t.Cleanup(func() {
+		if err := RemoveTestWorkspaces(ctx, client, config.Organization, workspacePrefix); err != nil {
+			t.Fatal(err)
+		}
+	})
+}
+
+func TestImportExistingResources(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	ctx := context.Background()
+
+	config := NewTestRunConfig()
+
+	config.Variables = `---
+- key: foo
+  value: baz
+  category: terraform`
+
+	client, err := tfe.NewClient(&tfe.Config{
+		Address: fmt.Sprintf("https://%s", config.Host),
+		Token:   config.Token,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err := RemoveTestWorkspaces(ctx, client, config.Organization, workspacePrefix); err != nil {
+		t.Fatal(err)
+	}
+
+	ws, err := client.Workspaces.Create(ctx, config.Organization, tfe.WorkspaceCreateOptions{
+		Name:             &config.Name,
+		TerraformVersion: strPtr("1.0.4"),
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, ws.TerraformVersion, "1.0.4")
+
+	v, err := client.Variables.Create(ctx, ws.ID, tfe.VariableCreateOptions{
+		Key:      strPtr("foo"),
+		Value:    strPtr("bar"),
+		Category: tfe.Category(tfe.CategoryTerraform),
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, v.Value, "bar")
+
+	Run(config)
+
+	ws, err = client.Workspaces.Read(ctx, config.Organization, config.Name)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, ws.TerraformVersion, "1.0.5")
+
+	v, err = client.Variables.Read(ctx, ws.ID, v.ID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, v.Value, "baz")
+
+	t.Cleanup(func() {
+		if err := RemoveTestWorkspaces(ctx, client, config.Organization, workspacePrefix); err != nil {
+			t.Fatal(err)
+		}
+	})
+}
+
+func TestDriftCorrection(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	ctx := context.Background()
+
+	config := NewTestRunConfig()
+
+	client, err := tfe.NewClient(&tfe.Config{
+		Address: fmt.Sprintf("https://%s", config.Host),
+		Token:   config.Token,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err := RemoveTestWorkspaces(ctx, client, config.Organization, workspacePrefix); err != nil {
+		t.Fatal(err)
+	}
+
+	ws, err := client.Workspaces.Create(ctx, config.Organization, tfe.WorkspaceCreateOptions{
+		Name:             &config.Name,
+		TerraformVersion: &config.TerraformVersion,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	v, err := client.Variables.Create(ctx, ws.ID, tfe.VariableCreateOptions{
+		Key:      strPtr("foo"),
```

Comment:
```suggestion
		Key:      tfe.String("foo"),
```

Packages that use a lot of scalar pointers, especially API clients, generally expose their own methods for returning a pointer for a value.

Reply:
Multiple instances of this, btw

Reply:
oh awesome! felt weird about writing my own here.